vllm/model_executor/layers/quantization/torchao_utils.py

```diff
@@ -0,0 +1,93 @@
+# SPDX-License-Identifier: Apache-2.0
+"""
+Common utilities for torchao.
+"""
+
+from typing import Dict, Set
+
+import torch
+
+
+def torchao_quantize_param_data(param: torch.Tensor, torchao_config: str):
+    """Quantize a Tensor with torchao quantization specified by torchao_config
+
+    Args:
+       `param`: weight parameter of the linear module
+       `torchao_config`: type of quantization and their arguments we want to
+        use to quantize the Tensor, e.g. int4wo-128 means int4 weight only
+        quantization with group_size 128
+    """
+    # Lazy import to suppress some warnings
+    from torchao.quantization import (float8_dynamic_activation_float8_weight,
+                                      int4_weight_only,
+                                      int8_dynamic_activation_int8_weight,
+                                      int8_weight_only, quantize_)
+    from torchao.quantization.observer import PerRow, PerTensor
```

Comment:
Can you wrap the torchao import in a try-except so we can give a nice error message instructing the user how to install torchao? See deepspeed as an example https://github.com/vllm-project/vllm/blob/084bbac8cc4c29b7dcd2098418168c61d3d42e9b/vllm/model_executor/layers/quantization/deepspeedfp.py#L144-L153